Advanced-Recruit-Training/src/I_BatteryStateOfChargeService.h

```diff
@@ -1,7 +1,8 @@
 #pragma once
 
 #include <QTime>
-struct BatteryData;
+#include "BatteryData.h"
```

Comment:
Typically, the implementation file (.cpp) should be the one to include other header files. 

That way, if a new class includes `I_BatteryStateOfChargeService.h`, it doesn't also accidentally grab `BatteryData.h`. 

From there, I believe `struct BatteryData` should work.